diff --git a/dist/queue.js b/dist/queue.js
index bcc4149479070c3bf84339cd62c4bd5534524597..95ac9710ded8f8884770c660cba7fe289727b85e 100644
--- a/dist/queue.js
+++ b/dist/queue.js
@@ -41,14 +41,16 @@ class Queue extends tiny_typed_emitter_1.TypedEmitter {
                 switch (this.loop.type) {
                     case LoopType.Song:
                         await this.options.play(this, this.current);
+                        this.emit("trackEnd", this.current, reason);
                         return;
                     case LoopType.Queue:
                         this.previous.push(this.current);
+                        this.tracks?.push(this.current);
                         break;
                     case LoopType.None:
                         break;
                 }
-                this.emit("trackEnd", this.current);
+                this.emit("trackEnd", this.current, reason);
             }
             if (!this.tracks.length) {
                 this.tracks = this.previous;
