diff --git a/node_modules/@lavaclient/queue/dist/Queue.js b/node_modules/@lavaclient/queue/dist/Queue.js
index 49cd399..c41414f 100644
--- a/node_modules/@lavaclient/queue/dist/Queue.js
+++ b/node_modules/@lavaclient/queue/dist/Queue.js
@@ -44,7 +44,7 @@ class Queue extends tiny_typed_emitter_1.TypedEmitter {
                         this.previous.push(this.current);
                         break;
                 }
-                this.emit("trackEnd", this.current);
+                this.emit("trackEnd", this.current, reason);
             }
             if (!this.tracks.length) {
                 this.tracks = this.previous;
