diff --git a/node_modules/@lavaclient/queue/dist/Queue.js b/node_modules/@lavaclient/queue/dist/Queue.js
index 49cd399..722872a 100644
--- a/node_modules/@lavaclient/queue/dist/Queue.js
+++ b/node_modules/@lavaclient/queue/dist/Queue.js
@@ -42,9 +42,10 @@ class Queue extends tiny_typed_emitter_1.TypedEmitter {
                         return this.player.play(this.current, {});
                     case LoopType.Queue:
                         this.previous.push(this.current);
+                        this.tracks?.push(this.current);
                         break;
                 }
-                this.emit("trackEnd", this.current);
+                this.emit("trackEnd", this.current, reason);
             }
             if (!this.tracks.length) {
                 this.tracks = this.previous;
