diff --git a/dist/index.d.ts b/dist/index.d.ts
index 50ca37d5254bee490408bf675255bb1e3d475e9f..74832c3ae75f13d17927b41a8d5e249ce63de6d0 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -124,7 +124,7 @@ type KeyvOptions = {
 type KeyvOptions_ = Omit<KeyvOptions, 'store'> & {
     store: KeyvStoreAdapter | Map<any, any> & KeyvStoreAdapter;
 };
-type IteratorFunction = (argument: any) => AsyncGenerator<any, void>;
+type IteratorFunction = () => AsyncGenerator<any, void>;
 declare class Keyv<GenericValue = any> extends EventManager {
     opts: KeyvOptions_;
     iterator?: IteratorFunction;
