diff --git a/dist/index.d.ts b/dist/index.d.ts
index fbfc1842f1f81c872305649195ba19c0000b4acb..96fce6908ce6ba7e07aa337051ac3a28248ebacd 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -161,7 +161,7 @@ type KeyvOptions = {
 type KeyvOptions_ = Omit<KeyvOptions, "store"> & {
     store: KeyvStoreAdapter | (Map<any, any> & KeyvStoreAdapter);
 };
-type IteratorFunction = (argument: any) => AsyncGenerator<any, void>;
+type IteratorFunction = () => AsyncGenerator<any, void>;
 declare class Keyv<GenericValue = any> extends EventManager {
     opts: KeyvOptions_;
     iterator?: IteratorFunction;
