diff --git a/package.json b/package.json
index 07f82a5b377c6a85a3e7712ea711e80c299b2c61..38d8ebfbd67c57fc09fd26a74a70e53ee320945f 100644
--- a/package.json
+++ b/package.json
@@ -95,5 +95,6 @@
   "lint-staged": {
     "src/**/*.js": "eslint",
     "test/**/*.js": "eslint"
-  }
+  },
+  "types": "./src/index.d.ts"
 }
diff --git a/src/index.d.ts b/src/index.d.ts
new file mode 100644
index 0000000000000000000000000000000000000000..0ed4462ee269fdff75693b6c828a79936a6a3fe6
--- /dev/null
+++ b/src/index.d.ts
@@ -0,0 +1,38 @@
+// to avoid mistakenly treating d.ts as js by ESLint.
+/* eslint-disable */
+declare class Kuroshiro {
+    constructor();
+    init(_analyzer: any): Promise<void>;
+    convert(
+        str: string,
+        options?: {
+            to?: 'hiragana' | 'katakana' | 'romaji';
+            mode?: 'normal' | 'spaced' | 'okurigana' | 'furigana';
+            romajiSystem?: 'nippon' | 'passport' | 'hepburn';
+            delimiter_start?: string;
+            delimiter_end?: string;
+        }
+    ): Promise<string>;
+    Util: {
+        isHiragana: (ch: string) => boolean;
+        isKatakana: (ch: string) => boolean;
+        isKana: (ch: string) => boolean;
+        isKanji: (ch: string) => boolean;
+        isJapanese: (ch: string) => boolean;
+        hasHiragana: (str: string) => boolean;
+        hasKatakana: (str: string) => boolean;
+        hasKana: (str: string) => boolean;
+        hasKanji: (str: string) => boolean;
+        hasJapanese: (str: string) => boolean;
+        kanaToHiragana: (str: string) => string;
+        kanaToKatakana: (str: string) => string;
+        kanaToRomaji: (
+            str: string,
+            system: "nippon" | "passport" | "hepburn"
+        ) => string;
+    };
+}
+
+declare module "kuroshiro" {
+    export = Kuroshiro;
+}
