diff --git a/node_modules/winston-loki/index.d.ts b/node_modules/winston-loki/index.d.ts
index c550bec..f794210 100644
--- a/node_modules/winston-loki/index.d.ts
+++ b/node_modules/winston-loki/index.d.ts
@@ -10,6 +10,7 @@ declare interface LokiTransportOptions extends TransportStream.TransportStreamOp
     json?: boolean;
     batching?: boolean;
     labels?: object;
+    format?: any;
     clearOnError?: boolean,
     replaceTimestamp?: boolean,
     gracefulShutdown?: boolean,
